mule-extensions-api-xml-dsl/src/main/java/org/mule/runtime/extension/xml/dsl/api/resources/spi/SchemaResourceFactory.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.xml.dsl.api.resources.spi;
+
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.extension.api.resources.GeneratedResource;
+
+/**
+ * Creates a {@link GeneratedResource} instance with the schema from an {@link ExtensionModel} instance.
+ * <p/>
+ * Implementations are to be reusable and thread-safe.
+ *
+ * @since 1.0
+ */
+public interface SchemaResourceFactory extends DslResourceFactory {
```

Comment:
@fernandezlautaro I think that the explanation you provide must be in the javadoc